test/jdk/javax/swing/JList/bug4618767.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4618767
+ * @summary First letter navigation in JList interferes with mnemonics
+ * @key headful
+ * @run main bug4618767
+ */
+
+import javax.swing.JFrame;
+import javax.swing.JList;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.SwingUtilities;
+import javax.swing.event.MenuEvent;
+import javax.swing.event.MenuListener;
+import java.awt.Robot;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.KeyEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class bug4618767 {
+    private static JFrame f;
+    private static JList list;
+    private static boolean menuSelected;
+    private static boolean failed;
+    private static CountDownLatch listGainedFocusLatch;
+    public static void main(String[] args) throws Exception {
+        try {
+            listGainedFocusLatch = new CountDownLatch(1);
+            createUI();
+            runTest();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (f != null) {
+                    f.dispose();
+                }
+            });
+        }
+    }
+
+    private static void createUI() throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            f = new JFrame("bug4618767");
+            JMenu menu = new JMenu("File");
+            menu.setMnemonic('F');
+            JMenuItem menuItem = new JMenuItem("item");
+            menu.add(menuItem);
+            JMenuBar menuBar = new JMenuBar();
+            menuBar.add(menu);
+            f.setJMenuBar(menuBar);
+
+            menu.addMenuListener(new MenuListener() {
+                public void menuCanceled(MenuEvent e) {}
+                public void menuDeselected(MenuEvent e) {}
+                public void menuSelected(MenuEvent e) {
+                    menuSelected = true;
+                }
+            });
+
+            list = new JList(new String[] {"one", "two", "three", "four"});
+            list.addFocusListener(new FocusAdapter() {
+                @Override
+                public void focusGained(FocusEvent e) {
+                    listGainedFocusLatch.countDown();
+                }
+            });
+            f.add(list);
+            f.pack();
+            f.setLocationRelativeTo(null);
+            f.setAlwaysOnTop(true);
+            f.setVisible(true);
+        });
+    }
+
+    private static void runTest() throws Exception {
+        if (!listGainedFocusLatch.await(3, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Waited too long, but can't gain" +
+                " focus for list");
+        }
+        Robot robot = new Robot();
+        robot.setAutoDelay(500);
```

Comment:
Can we reduce autoDelay to a smaller number?

Reply:
I was seeing some issues related to autoDelay in our CI, so used 500ms.
I think it is resolved and we can keep it less.
Updated to 200ms.